### DISCUSSION TITLE
Accessing page config centrally

### DISCUSSION THREAD

**Discussion by controversial** [vercel/next.js#16741] [Ideas]
Next.js supports exporting a `config` option from page files in order to access some features like enabling AMP or disabling javascript:
```js
export const config = { amp: true }
```
In several Next.js apps I've written, I've used a pattern of attaching properties to page component functions in order to create “internal” config options, which I then access in `_app.js`. For example, a configuration option to hide a footer on some pages would look like:
```js
// pages/index.js
export default function MyPage() { ... }
MyPage.hideFooter = true
```
```jsx
// pages/_app.js
export default function AppContainer({ Component, pageProps }) {
  return (
    <>
      <Component {...pageProps} />
      {!Component.hideFooter && <Footer />}
    </>
  );
}
```

If there were a way for the `_app` page component to read from the exported page config that Next.js uses for internal configuration options, we could write more expressive code, like
```js
// pages/index.js
export default function MyPage() { ... }
export const config = { hideFooter: true }
```

Does Next expose this exported `config` object at runtime anywhere where the `_app` component could get it?

**Comment by timneutkens** [comment]
> Does Next expose this exported config object at runtime anywhere where the _app component could get it?

It is not exposed and should not have custom properties set on it. It's extracted out and in the future it'll potentially be tree shaken after extracting the data to save bundle size.
